src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -246,6 +290,49 @@ internal static class PeWriter
             var peBlob = new BlobBuilder();
             var peContentId = peBuilder.Serialize(peBlob, out Blob mvidSectionFixup);
 
+            if (SRMVisibleMachine != properties.Machine)
+            {
+                // ARM64 pe
+                // Update Amd64 machine type in PE header to 0xAA64
+                // Machine type in PEHeader is located at...
+                // 128 bytes (dos header)
+                // 4 bytes (PE Signature)
+                // 2 bytes (Machine)
+                // 4 bytes (TimeStamp)
+                ushort amd64MachineType = (ushort)Machine.Amd64;
+                Debug.Assert(ReadByteFromBlobBuilder(peBlob, 128+4) == (byte)(amd64MachineType));
+                Debug.Assert(ReadByteFromBlobBuilder(peBlob, 128+4 + 1) == (byte)(amd64MachineType >> 8));
+
+                ushort realMachineType = (ushort)properties.Machine;
+                byte lsbMachineType = (byte)realMachineType;
+                byte msbMachineType = (byte)(realMachineType >> 8);
+
+                // This code path is currently only expected to be used for ARM64
+                Debug.Assert(lsbMachineType == 0x64);
+                Debug.Assert(msbMachineType == 0xAA);
+
+                WriteByteToBlobBuilder(peBlob, 128+4, lsbMachineType);
+                WriteByteToBlobBuilder(peBlob, 128+4 + 1, msbMachineType);
+
+                if (peIdProvider != null)
```

Comment:
You may want to test this. There are some tests for determinism /w platform switches - like `CompareAllBytesEmitted_Release`

Reply:
Done... and it found bugs so thanks for the pointer to where those tests are.